llvm/lib/Transforms/Scalar/LoopInterchange.cpp

```diff
@@ -156,7 +158,10 @@ static bool populateDependencyMatrix(CharMatrix &DepMatrix, unsigned Level,
           Dep.push_back('I');
         }
 
-        DepMatrix.push_back(Dep);
+        // Make sure we only add unique entries to the dependency matrix.
+        if (Seen.insert(std::string(Dep.begin(), Dep.end())).second) 
```

Comment:
```suggestion
        if (Seen.insert(StringRef(Dep.data(), Dep.size())).second) 
```
Doesn't something like this work?

Reply:
Thanks, will check and apply this before committing this, which I assume is okay.